_source/_posts/2022-09-26-micro-frontends-java-microservices.adoc

```diff
@@ -0,0 +1,625 @@
+. Changed the service discovery to https://www.consul.io/[Consul].^1^
+. Added a deployment section for Kubernetes.
+
+^1^ We'll default to Consul in JHipster v8.
+
+CAUTION: JHipster's JDL allows you to specify different client frameworks, but micro frontends will only work if you specify the same one for all your apps.
+
+Run the following command to create a new directory for your micro frontends project:
+
+[source,shell]
+----
+take micro-frontends-jhipster
+# mkdir micro-frontends-jhipster && cd micro-frontends-jhipster if you don't have take installed
+----
+
+Download both the `-ms` and `-mf` JDLs and compare them in IntelliJ IDEA. You'll need IDEA's https://www.jetbrains.com/help/idea/working-with-the-ide-features-from-command-line.html[Command-line Launcher] for the `idea` command to work.
+
+[source,shell]
+----
+jhipster download reactive-ms.jdl
+jhipster download reactive-mf.jdl
+idea diff reactive-ms.jdl reactive-mf.jdl
+----
+
+image::{% asset_path 'blog/micro-frontends-java/jdl-ms-vs-mf-diff.png' %}[alt=Reactive microservices refactored to micro frontends,width=800,align=center,link={% asset_path 'blog/micro-frontends-java/jdl-ms-vs-mf-diff.png' %}]
+
+image::{% asset_path 'blog/micro-frontends-java/jdl-with-kubernetes.png' %}[alt=Additional K8s deployment,width=800,align=center,link={% asset_path 'blog/micro-frontends-java/jdl-with-kubernetes.png' %}]
+
+TIP: The https://plugins.jetbrains.com/plugin/19697-jhipster-jdl[JHipster JDL Plugin] is a handy tool for working with JDL files.
+
+=== Micro frontend options: Angular, React, and Vue
+
+JHipster has support for the big three JavaScript frameworks: Angular, React, and Vue. All are implemented using TypeScript and a newly generated app should have around 70% code coverage, both on the backend and frontend.
+
+There is also a https://github.com/jhipster/generator-jhipster-svelte[Svelte blueprint], but it does not support micro frontends at the time of this writing.
+
+=== Build Java microservices with Spring Boot and WebFlux
+
+To generate a microservices architecture with micro frontend support, run the following command:
+
+[source,shell]
+----
+jhipster jdl reactive-mf.jdl --monorepository --workspaces
+----
+
+The last two arguments are optional, but I expect you to use them for this tutorial. Without the `monorepository` flag, the gateway and microservices would have their own Git repos. The `workspaces` flag enables https://docs.npmjs.com/cli/v8/using-npm/workspaces[npm workspaces], which are kinda like having an aggregator `pom.xml` that allows you to execute commands across projects. It also makes it so there's only one `node_modules` in the root directory. To learn more, I recommend egghead's https://egghead.io/courses/introduction-to-monorepos-with-npm-workspaces-c03f500b[Introduction to Monorepos with NPM Workspaces].
+
+If you want to use Angular, you can pass in `--client-framework angularX` to override the JDL value. For example:
```

Comment:
I think just `angular` should also work already as we list that as acceptable in the docs

Reply:
Nope. There's an error when I run it:

```
✖ An error occured while running jhipster:client#microfrontend
ERROR! Client framework angular doesn't support microfrontends
```